Source/missiles.cpp

```diff
@@ -393,9 +393,12 @@ void CheckMissileCol(Missile &missile, int minDamage, int maxDamage, bool isDama
 	if (mid > 0 || (mid != 0 && Monsters[abs(mid) - 1].mode == MonsterMode::Petrified)) {
 		mid = abs(mid) - 1;
 		if (missile.IsTrap()
-		    || (missile._micaster == TARGET_PLAYERS
-		        && (Monsters[missile._misource].flags & MFLAG_TARGETS_MONSTER) != 0
-		        && ((Monsters[mid].flags & MFLAG_GOLEM) != 0 || (Monsters[missile._misource].flags & MFLAG_BERSERK) != 0))) {
+		    || (missile._micaster == TARGET_PLAYERS                                 // or was fired by a monster
+		        && (Monsters[missile._misource].flags & MFLAG_TARGETS_MONSTER) != 0 // which can target other monsters

```

Comment:
the flag only indicates the type of the current target, not what the monster will possibly target
```suggestion
		        && (Monsters[missile._misource].flags & MFLAG_TARGETS_MONSTER) != 0 // which is target other monsters
```